### PR TITLE
feat: add Mongo DB types for bot usage

### PR DESCRIPTION
**Pull request by tcoratger**
### Feature Addition: MongoDB Types for Bot Utilization

This pull request introduces a set of MongoDB types tailored specifically for bot usage within the application. These types encapsulate essential document structures related to transactions, user details, rewards, swaps, and vesting. These types will enhance code readability, maintainability, and assist in ensuring robust typing within the MongoDB ecosystem.

### Added Types:
- `MongoTransfer`: Represents transfer transactions in MongoDB.
- `MongoUser`: Represents user details stored in MongoDB.
- `MongoReward`: Documents reward transactions in the MongoDB structure.
- `MongoSwap`: Captures details of swap transactions in the MongoDB schema.
- `MongoVesting`: Encompasses vesting details within the MongoDB structure.

### CODE REVIEW

**Review by juniormp — APPROVED** [review]

File: src/bot/constants.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Different transaction statuses.
+ */
+export const TRANSACTION_STATUS = {
```
Comment (milukove) [inline src/bot/constants.ts]: @tcoratger Maybe will be clearer to use enum here? Example: https://github.com/grindery-io/grindery-wallet/blob/main/src/types/State.ts#L94-L100

Here is why I think enum is clearer: https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums

**Review by milukove — APPROVED** [review]
@tcoratger  Looks good, thanks.

As a future improvement I would propose to extract repeating properties as its own type. 

For example we have a lot of properties that contain telegram id:
* userTelegramID
* senderTgId
* etc...

We could have `export type TelegramUserId = string;`, and then reuse this type for all fields containing tg id.

Same for chainId, token addresses, amounts, etc.

This way we will be sure that parameter type is the same across different objects. Helps to avoid mistakes were in one object some property is a string, and in the other object the same property is number.

File: src/bot/constants.ts (hunk 1 above)
Comment (tcoratger) [reply src/bot/constants.ts]: @milukove Fully aligned, addressed here: https://github.com/grindery-io/grindery-bot-auth/pull/187

**Comment by tcoratger** [comment]
> @tcoratger Looks good, thanks.
> 
> As a future improvement I would propose to extract repeating properties as its own type.
> 
> For example we have a lot of properties that contain telegram id:
> 
> * userTelegramID
> * senderTgId
> * etc...
> 
> We could have `export type TelegramUserId = string;`, and then reuse this type for all fields containing tg id.
> 
> Same for chainId, token addresses, amounts, etc.
> 
> This way we will be sure that parameter type is the same across different objects. Helps to avoid mistakes were in one object some property is a string, and in the other object the same property is number.

Agreed, I've addressed it also in a commit here to be merged in the same PR.

**Pull request merged**